src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Metadata/Models/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement.Metadata.Settings;
+
+namespace OrchardCore.ContentManagement.Metadata.Models
+{
+    public static class ContentPartFieldDefinitionExtensions
+    {
+        public static bool IsNamedPart(this ContentPartFieldDefinition fieldDefinition)
+            => fieldDefinition.PartDefinition.IsReusable() && fieldDefinition.ContentTypePartDefinition.Name != fieldDefinition.PartDefinition.Name;
+
+        /// <summary>
+        /// Returns full field name.
+        /// </summary>
+        /// <param name="fieldDefinition">The <see cref="ContentPartFieldDefinition" />.</param>
+        /// <returns></returns>
+        public static string GetFullName(this ContentPartFieldDefinition fieldDefinition)
+            => $"{fieldDefinition.PartDefinition.Name}-{fieldDefinition.Name}";
+
+        /// <summary>
+        /// Returns unique full field name.
+        /// </summary>
+        /// <param name="fieldDefinition">The <see cref="ContentPartFieldDefinition" />.</param>
+        /// <returns></returns>
+        public static string GetUniqueFullName(this ContentPartFieldDefinition fieldDefinition)
```

Comment:
This returns a css class and not a full name. Maybe GetReusableFieldWrapperClassName and the other one GetFieldWrapperClassName